packages/cli/src/commands/upgrade.js

```diff
@@ -404,10 +399,10 @@ const dedupeDeps = async (task, { verbose }) => {
     if (yarnVersion > 1) {
       await execa('yarn', ['dedupe'], baseExecaArgsForDedupe)
     } else {
-      await execa(
-        'npx',
-        [...npxArgs, 'yarn-deduplicate'],
-        baseExecaArgsForDedupe,
+      // Redwood projects should not be using yarn 1.x as we specify a version of yarn in the package.json
+      // with "packageManager": "yarn@4.1.1" or similar.
+      task.skip(
+        "Yarn 1.x doesn't support dedupe directly. Please use `npx` and run `npx yarn-deduplicate` manually.",
```

Comment:
I'm a little worried someone (maybe me!) will see this in a few months/years and think "Ohh, we can be more helpful here and just run npx for them!" and then go ahead and reintroduce the code you just got rid of.

Reply:
We could try running `npx yarn-deduplicate` maybe, and then, if that doesn't work we could install yarn-deduplicate locally and run it with `yarn yarn-deduplicate`. And if all of that doesn't work we could just tell the user to upgrade to yarn v4.

Or we skip trying to be nice and just tell them to install yarn v4 and then run `yarn dedupe` manually.

Either way, a comment here about npx maybe not being available on the user's system is probably a good idea (as I hinted at in my first comment)

Reply:
Just updated the comment to include why we do not want to do this automatically ([db4b9c4](https://github.com/redwoodjs/redwood/pull/10479/commits/db4b9c41db874c42a5f33de0797df3b1f08f86a0))

I don't have any particular metrics to back this up but I would expect almost all redwood users to be using yarn v4 as it is specified in the `package.json`. People upgrading projects might not though. I agree we could try, then install, and then try again to do it automatically but I would say we should avoid that for now unless it becomes clear a number of people are running into it and it's a bad dx.

Reply:
> I would say we should avoid that for

I agree. But if we can't run `npx`, most likely the user won't be able to either. So maybe tell them to upgrade to yarn 4 instead of suggesting something that most likely will fail? What do you think?

Reply:
We no longer even try to use `npx` so it could be available although now we don't require it to be.

Happy to switch the message to be upgrade `yarn` rather than use `npx`